test/evmc/mocked_host.hpp

```diff
@@ -49,7 +48,7 @@ struct MockedAccount
     uint256be balance;
 
     /// The account storage map.
-    std::map<bytes32, storage_value> storage;
+    std::unordered_map<bytes32, storage_value> storage;
```

Comment:
@bshastry would this cause any issue to the fuzzer?

Reply:
Yes. The main thing is that it would make the order of storage slots unpredictable, and hence show up differences in unoptimised and optimised code even though contents are the same (more context here: https://github.com/ethereum/solidity/pull/11094)

Reply:
Thanks! Applied the change here and documented it in the README.